src/NuGet.Core/NuGet.Packaging/Signing/Utility/RepositorySignatureInfoUtility.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using NuGet.Common;
+using NuGet.Packaging.Core;
+
+namespace NuGet.Packaging.Signing
+{
+    public static class RepositorySignatureInfoUtility
+    {
+        /// <summary>
+        /// Gets SignedPackageVerifierSettings from a given RepositorySignatureInfo. 
+        /// </summary>
+        /// <param name="repoSignatureInfo">RepositorySignatureInfo to be used.</param>
+        /// <param name="commonSignedPackageVerifierSettings">SignedPackageVerifierSettings to be used if RepositorySignatureInfo is unavailable.</param>
+        /// <returns>SignedPackageVerifierSettings based on the RepositorySignatureInfo and SignedPackageVerifierSettings.</returns>
+        public static SignedPackageVerifierSettings GetSignedPackageVerifierSettings(
+            RepositorySignatureInfo repoSignatureInfo,
+            SignedPackageVerifierSettings commonSignedPackageVerifierSettings)
```

Comment:
Can you just name this `defaultSettings` or `fallbackSettings`?  Update XML doc comments too.